File: stack_1/server/routes/messages.py
from fastapi import APIRouter
from database.db_find import find_messages
from database.db_insert import insert_message
import datetime

router = APIRouter()


@router.get("/messages")
async def get_messages():
    messages = await find_messages()
    return messages


@router.post("/messages")
async def post_message(message: dict):
    message["timestamp"] = datetime.datetime.now(datetime.UTC).isoformat()
    result = await insert_message(message)
    message["_id"] = str(result.inserted_id)
    return message

File: stack_1/client/src/apis/messages.js
import axios from "axios";

const API_URL = "http://127.0.0.1:5000/api";

export const getMessages = async () => {
	const response = await axios.get(`${API_URL}/messages`);
	return response.data;
};

export const createMessage = async (message) => {
	const response = await axios.post(`${API_URL}/messages`, message);
	return response.data;
};
